dependencies.yaml

```diff
@@ -71,6 +71,11 @@ dependencies:
               cuda: "11.8"
             packages:
               - cudatoolkit=11.8
```

Comment:
I think this should be updated to use `cuda-version` as the constraint. Similar to https://github.com/rapidsai/rmm/pull/1296/files#diff-5475a6d76de4c506ee92cf6f941bba30a6a07b5881cfea531d42a6ec035095a6

Not a blocker though

Reply:
Yes, please adopt changes like those in the linked RMM PR. See also this cuDF example: https://github.com/rapidsai/cudf/pull/13615

Basically there should never be any pinnings on `cudatoolkit`. Its version should be controlled by pinnings on `cuda-version`.

Reply:
Done in https://github.com/rapidsai/ucx-py/pull/967/commits/0e5b0c2a88f31873c1aabcb0ec268d7ab6cff28a .